### ISSUE TITLE
[V0.1]  Provide PROD dataset to manually set ITF date for in-motion pension applications

### ISSUE THREAD

**Issue by matt4su** [department-of-veterans-affairs/va.gov-team#81413]
**NOTE: this ticket should be done AFTER we've launched ITF V0**

Taking this step will help ensure the Veteran with in-motion claims receive the earliest possible start date for their benefit.

# Order
#90364 - First because it addresses the overall issue and should prevent future cases
#90176 - This ticket sets the participant_id for all of the pension in_progress veterans.
#81413 - (this one) - Last because it is the remediation for the overall issue

**Tasks:**
* Post V0 launch, pull dataset containing:
1. Utilize data format and fields finalized in #84670
2. Pull complete data set between 1/31 and <ITF 100% date>

* Review dataset with product -> send file to VBA (work with Sanja)
* PII to be shared via Sharepoint

**AC**
- No PII violations (No PII on dev machines. Use Sharepoint to transmit PII.)
- Receive confirmation that batch was run successfully (or iterate until confirmation is received)

**Comment by mjknight50** [comment]
This is a separate task that can be done at anytime after we complete the main ITF work.

**Comment by sanjabaj2** [comment]
When do we plan to share this file? Also, I don't have prod access, how will the file be shared with me?

**Comment by mjknight50** [comment]
Hi Sanja,
We first need to do this ticket:
https://github.com/department-of-veterans-affairs/va.gov-team/issues/90176

This will try to create a participant_id for any that are missing one.  Once that is done, we can do the ITF export.  We can post on sharepoint if that works?

**Comment by sanjabaj2** [comment]
> Hi @sanjabaj2 , We first need to do this ticket: #90176
> 
> This will try to create a participant_id for any that are missing one. Once that is done, we can do the ITF export. We can post on sharepoint if that works?

Ideally, but do we know if MPI had added the code to check for duplicates? Also, I thought we said LH will do this step when needed. In my mind, this was future implementation that we will backfill when functionality is available.

**Comment by mjknight50** [comment]
This is meant to be a one-time independent run to create the participant_id.  It is not affiliated with the other long slack thread about how to fix this long-term.    This is just a one-off fix for the veterans associated with the pension form.
If you prefer we skip this, we can, but I assume that when we pass over the ICN's, any without a participant_id will fail when they try to manually set the ITF.

**Comment by mjknight50** [comment]
The decision was made to run this now and not wait for the participant_id fix:
https://github.com/department-of-veterans-affairs/va.gov-team/issues/90176

https://dsva.slack.com/archives/C05MS96P6P2/p1725643055150279

We'll need to run this script and create the excel list twice, so it is important to note the date/time of the export so that the next time it is run, it will be easy for someone to run it again.

**Comment by sanjabaj2** [comment]
> The decision was made to run this now and not wait for the participant_id fix: #90176
> 
> https://dsva.slack.com/archives/C05MS96P6P2/p1725643055150279
> 
> We'll need to run this script and create the excel list twice, so it is important to note the date/time of the export so that the next time it is run, it will be easy for someone to run it again.

Yes, and to make sure we don't duplicate same records in follow up file. Perhaps, we can ask for results list after the initial run and update our side? Just asking if that makes sense?
